electron/wine/runtimes/util.ts

```diff
@@ -0,0 +1,86 @@
+import axios from 'axios'
+import { spawn } from 'child_process'
+import { mkdirSync, writeFileSync } from 'graceful-fs'
+import { logError, LogPrefix } from './../../logger/logger'
+
+interface GithubAssetMetadata {
+  url: string
+  id: number
+  node_id: string
+  name: string
+  label: string
+  content_type: string
+  state: string
+  size: number
+  download_count: number
+  created_at: string
+  updated_at: string
+  browser_download_url: string
+}
+
+/**
+ * Takes a GitHub download URL (https://www.github.com/AUTHOR/REPO/releases/download/TAGNAME/ASSET) and returns relevant asset metadata
+ * @param url
+ * @returns
+ */
+async function getAssetDataFromDownload(
+  url: string
+): Promise<GithubAssetMetadata> {
+  const splitUrl = url.split('/').filter(Boolean)
+  const [, , author, repo, , , tag, assetName] = splitUrl
+  const releaseData = (
+    await axios.get(
+      `https://api.github.com/repos/${author}/${repo}/releases/tags/${tag}`
+    )
+  ).data
+  const assets: Array<GithubAssetMetadata> = releaseData.assets
+  return assets.find((asset) => {
```

Comment:
Check that assets is not undefined, else you can get `find is not a function of undefined`